src/components/data/RenameColumnModal.js

```diff
@@ -0,0 +1,55 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { FormLabel } from 'src/libs/forms'
+import * as Utils from 'src/libs/utils'
+
+
+export const tableNameInput = ({ inputProps, ...props }) => h(ValidatedInput, {
+  ...props,
+  inputProps: {
+    ...inputProps,
+    autoFocus: true,
+    placeholder: 'Enter a name'
+  }
+})
+
+const RenameColumnModal = ({ onDismiss, onSuccess, namespace, name, entityType, oldAttributeName }) => {
+  // State
+  const [newAttributeName, setNewAttributeName] = useState('')
+  const [renaming, setRenaming] = useState(false)
+
+  return h(Modal, {
+    onDismiss,
+    title: 'Rename Column',
+    okButton: h(ButtonPrimary, {
+      disabled: renaming,
+      onClick: _.flow(
+        withErrorReporting('Error renaming column name.'),
+        Utils.withBusyState(setRenaming)
```

Comment:
Using the `Utils.withBusyState` wrapper, if the rename request is successful, `setRenaming(false)` is called after the `onSuccess` callback. Since `onSuccess` dismisses the modal, this results in `setRenaming(false)` causing a React warning about a state update on an unmounted component.

The way I've chosen to avoid this in some other components (SingleEntityEditor, AddColumnModal, etc) is to call `setBusy` and `reportError` directly. Alternatively, delaying `onSuccess` by wrapping it in `setTimeout(..., 0)` would probably work as well.
https://github.com/DataBiosphere/terra-ui/blob/c2f2ef23bffaf3f0c215638a3fb25b50ecbae5c2/src/components/data/data-utils.js#L1075-L1106

Reply:
Addressed in 3612f06503e935b2c9f198fae6defb15f278b55b